app/main/download_data.py

```diff
@@ -6,15 +6,35 @@
 from config import Config
 
 
-def quarter_to_date(quarter, year):
-    # January-March is Q1, April-June is Q2, July-September is Q3, and October-December is Q4
+def financial_quarter_from_mapping(quarter, year):
+    """
+
+    Args:
+        quarter: int - financial quarter selected by the user
+        year: str - financial year selected by the user
 
+    Returns:
+        Datetime stamp corresponding with the start of the chosen financial period
```

Comment:
presumably this returns Date stamp or None, as it is using a .get method?

Reply:
It's not possible for the user to leave this field blank in the UI, so None will never be returned even though it's using .get

Reply:
Ok, fiar enough, but in terms if function level docstrings, we either:
1. Put in the type hints / docstring for None, as from a function point of view it is entirely possible - the function should not know about anything outside the function. If some other function came along an used this one, and passed it params not in that dict (like 5) it would return None.
2. Switch it to a normal index lookup.

I think option 1 is better, it's cleaner

Reply:
returns str | None